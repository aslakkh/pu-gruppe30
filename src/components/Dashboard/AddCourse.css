.AddCourse{
    display: flex;
    flex-direction: column;
    align-items: center;
}

.AddCourseForm{
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: flex-start;
    margin-top: 10px;
    
}



