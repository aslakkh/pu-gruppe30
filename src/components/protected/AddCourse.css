.AddCourse{
    display: flex;
    flex-direction: column;
    align-items: center;
}

.FormWrapper{
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    margin-top: 10px;
}

