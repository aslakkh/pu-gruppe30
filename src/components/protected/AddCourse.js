import React, { Component } from 'react'
import { saveCourse,getCourse } from '../../helpers/auth'
import {Forms, FormGroup, FormControl, ControlLabel} from 'react-bootstrap'

//Component for adding AddCourse
//https://react-bootstrap.github.io/components.html#forms

export default class AddCourse extends Component{
    constructor(props){
        super(props);
        this.handleAdd = this.handleAdd.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.state = {
            value: ''
        }
    }

    //get intialstate of form component
    getInitialState(){
        return {
            value: ''
        }
    }

    //handle change in form value
    handleChange(e){
        this.setState({value: e.target.value});
        console.log(this.state.value);
    }

    //function for validating 
    getValidationState(){
        return 'success';
    }

    //fuction for handling click on add-button
    handleAdd = (e) => {
		e.preventDefault();
		saveCourse("TMA4245")
		console.log("hei")
		getCourse();
	}

    render(){
        return(
            <div className="AddCourse">
                <div className ="Form">
                    <form>
                        <FormGroup controlId="1" bsSize="small" style={{width: "50%"}}>
                            <FormControl 
                            id="1" value={this.state.value} 
                            type="text" 
                            placeholder="Course name"
                            onChange={this.handleChange}
                            />
                        </FormGroup>
                    </form>
                    <button className="btn btn-primary btn-lg">legg til emne</button>
                </div>
            </div>
        )
    } 


}