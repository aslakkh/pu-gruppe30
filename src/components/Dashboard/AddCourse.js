import React, { Component } from 'react'
import { saveCourse } from '../../helpers/auth'
import {FormGroup, FormControl, Button, HelpBlock} from 'react-bootstrap'
import './AddCourse.css'

/*      Component for adding course

Returns a form and a button.
Saves form data in firebase upon button click.
Should be updated to include better form validation. User should probably only be able to add courses that are already in database.

//https://react-bootstrap.github.io/components.html#forms

TODO

- prevent adding an already existing course from replacing previous course
- make it possible to submit form by pressing enter

*/


export default class AddCourse extends Component{
    constructor(props){
        super(props);
        this.handleAdd = this.handleAdd.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.state = {
            value: ''
        }
    }

    //get intialstate of form component. 
    getInitialState(){
        return {
            value: ''
        }
    }

    //handle change in form value. Sets the state value to form input value
    handleChange(e){
        this.setState({value: e.target.value.toUpperCase()});
        
    }

    //function for handling keypress
    handleKeyPress(e){
        if(e.key === 'Enter'){
            this.handleAdd();
        }
    }

    //function for validating input. For now returns error if form has input of length greater than 0 but less than 5, and success for anything of length > 4
    getValidationState(){
        var length = this.state.value.length;
        var re = /^[A-Z]{2,3}[0-9]{4}$/

        if(re.test(this.state.value)){
            return 'success';
        }else if(length > 0){
            return 'error';
        }
    }

    //function for handling click on add-button. Only saves course to database if input is valid.
    handleAdd = (e) => {
		e.preventDefault(); //prevents default browser behaviour on click, whatever that means

        if(this.getValidationState() === 'success'){
            saveCourse(this.state.value); //saveCourse(course) adds course to currently logged in user in Firebase
            this.setState({value: ''}); //resets value to empty string
        }
        else{
            console.log("Error: Tried submitting invalid value: " + this.state.value);
        }
		//getCourse();
	}

    //Renders a form and a button for submission
    render(){
        return(
            <div className="FormWrapper" >
                    <form>
                        <FormGroup controlId="addCourseForm" validationState={this.getValidationState()}>
                            <FormControl 
                            id="addCourseForm" 
                            componentClass='input'
                            value={this.state.value}
                            className="InputForm" 
                            type="text" 
                            placeholder="Course name"
                            onChange={this.handleChange}
                            />
                            <FormControl.Feedback/>
                            <HelpBlock>Enter course code (up to three letters followed by four digits)</HelpBlock>
                        </FormGroup>
                    </form>
                    <Button className="AddCourse-addBtn "type='submit' bsStyle="primary" bsSize="large" onClick={this.handleAdd} style={{marginLeft: "20px"}}>Add</Button>
            </div>
        )
    } 


}