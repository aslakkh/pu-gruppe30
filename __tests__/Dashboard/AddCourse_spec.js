import React from 'react';
import renderer from 'react-test-renderer';
import AddCourse from '../../src/components/Dashboard/AddCourse.js';
import shallow from 'enzyme'
import TestUtils from "react-addons-test-utils"
describe('AddCourse renders FormGroup and button', function() {
/*    const component = renderer.create(
        <AddCourse />
    ).toJSON();
    expect(component).toMatchSnapshot();*/

    let renderer2;
    let component2;
    renderer2 = TestUtils.createRenderer();
    renderer2.render(
        <AddCourse/>
    );
    component2 = renderer2.getRenderOutput();
    it("One course", () => {
        expect(component2).toMatchSnapshot();
    });
    it('addsession', () => {
        const wrapper = renderer(<AddCourse/>);
        wrapper.find('Button').simulate('click');



    })
});