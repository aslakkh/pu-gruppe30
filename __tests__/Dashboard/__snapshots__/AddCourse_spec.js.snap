// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCourse renders FormGroup and button One course 1`] = `
<div>
  <div
    style={
      Object {
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginTop": "10px",
      }
    }
  >
    <FormGroup
      bsClass="form-group"
      controlId="addCourseForm"
      validationState={undefined}
    >
      <FormControl
        bsClass="form-control"
        componentClass="input"
        id="addCourseForm"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Course name"
        type="text"
        value=""
      />
      <FormControlFeedback
        bsClass="form-control-feedback"
        bsRole="feedback"
      />
      <HelpBlock
        bsClass="help-block"
      >
        Enter course code (up to three letters followed by four digits)
      </HelpBlock>
    </FormGroup>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsSize="large"
      bsStyle="primary"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      Add
    </Button>
  </div>
  <FeedbackMessage
    active={undefined}
    bsStyle={undefined}
    message={undefined}
  />
</div>
`;
